libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/BotFrameworkHttpClient.cs

```diff
@@ -42,51 +54,61 @@ public class BotFrameworkHttpClient : BotFrameworkClient
             IChannelProvider channelProvider = null,
             ILogger logger = null)
         {
-            HttpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
-            CredentialProvider = credentialProvider ?? throw new ArgumentNullException(nameof(credentialProvider));
-            ChannelProvider = channelProvider;
-            Logger = logger ?? NullLogger.Instance;
-            ConnectorClient.AddDefaultRequestHeaders(HttpClient);
+            if (credentialProvider == null)
+            {
+                throw new ArgumentNullException(nameof(credentialProvider));
+            }
+
+            _httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
+            _logger = logger ?? NullLogger.Instance;
+
+            ConnectorClient.AddDefaultRequestHeaders(_httpClient);
+
+            OriginatingAudience = channelProvider != null && channelProvider.IsGovernment() ? GovernmentAuthenticationConstants.ToChannelFromBotOAuthScope : AuthenticationConstants.ToChannelFromBotOAuthScope;
+
+            _buildCredentialsAsyncImpl = async (appId, oAuthScope) =>
+            {
+                var appPassword = await credentialProvider.GetAppPasswordAsync(appId).ConfigureAwait(false);
+                return channelProvider != null && channelProvider.IsGovernment() ? new MicrosoftGovernmentAppCredentials(appId, appPassword, _httpClient, _logger, oAuthScope) : new MicrosoftAppCredentials(appId, appPassword, _httpClient, _logger, oAuthScope);
+            };
         }
 
         /// <summary>
-        /// Gets the Cache for appCredentials to speed up token acquisition (a token is not requested unless is expired).
-        /// AppCredentials are cached using appId + scope (this last parameter is only used if the app credentials are used to call a skill).
+        /// Initializes a new instance of the <see cref="BotFrameworkHttpClient"/> class.
         /// </summary>
-        /// <value>ConcurrentDictionary of <see cref="AppCredentials"/>.</value>
-        protected static ConcurrentDictionary<string, AppCredentials> AppCredentialMapCache { get; } = new ConcurrentDictionary<string, AppCredentials>();
```

Comment:
Can't delete protected member of public class - users may have inherited this class and accessed this protected member, this change would break them.

Reply:
Reverted this change.